mix.exs

```diff
@@ -52,7 +52,7 @@ defmodule JOSE.Mixfile do
       {:thoas, "~> 1.0", only: [:dev, :test]},
       {:ex_doc, "~> 0.30", only: :dev},
       {:earmark, "~> 1.4", only: :dev},
-      {:dialyxir, "~> 1.4.3"}
+      {:dialyxir, "~> 1.4.3", only: :dev}
```

Comment:
Thanks for creating this PR, I'm having a similar error at the moment. I'm going to use `override` while this being fix
![image](https://github.com/potatosalad/erlang-jose/assets/6057298/e528ab9a-d81c-4ca3-8f47-65fb0d4fcece)

Also should the `test` environment be added to the `only` option as describe in the doc?
![image](https://github.com/potatosalad/erlang-jose/assets/6057298/0f3a0e8e-49db-4d48-953e-ffec2c05ef63)

I'm not sure if having only `dev` will manage to fix the `Dependencies have diverged` error

Reply:
> I'm not sure if having only dev will manage to fix the Dependencies have diverged error

Not sure, I added the `runtime: false` flag just to be sure.